### DISCUSSION TITLE
Welcome to TPrintPreview-for-Delphi Discussions!

### DISCUSSION THREAD

**Discussion by landrix** [landrix/TPrintPreview-for-Delphi#5] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by JosBeukers** [comment]
In Windows 10 you can scale each display different.
I patched TPrintPreview.CreateMetaFileCanvas to support this new scaling.
Instead of using de mainscreen I use the FPaperview.Canvas.

I changed the original code:

```Delphi
procedure TPrintPreview.CreateMetafileCanvas(out AMetafile: TMetafile;
  out ACanvas: TCanvas);
begin
  AMetafile := TMetafile.Create;
  try
    with ScaleToDeviceContext(ReferenceDC, FDeviceExt) do
    begin

Replacement:

procedure TPrintPreview.CreateMetafileCanvas(out AMetafile: TMetafile;
  out ACanvas: TCanvas);
var aDC:HDC;
begin
  AMetafile := TMetafile.Create;
  try
    aDC:=ReferenceDC;
    if aDC=0 then
      aDC:=FPaperView.Canvas.Handle;
    with ScaleToDeviceContext(aDC, FDeviceExt) do
    begin
```
Hope this helps someone.

Regards 
Jos Beukers

**Reply by landrix** [reply]
i've added it
